beancount/utils/errors.cc

```diff
@@ -22,13 +23,21 @@ absl::StatusCode kSystemError = absl::StatusCode(kUserErrors + 1);
 
 absl::Status SystemError(absl::string_view message) {
   char buffer[2048];
-#ifndef __APPLE__
-  description = strerror_r(errno, buffer, 2048);
+
+#ifdef __GLIBC__
+  // GNU libc provides non-XSI-compliant strerror_r.
```

Comment:
Accordingly to the man page https://linux.die.net/man/3/strerror_r GNU libc can be made to use the XSI-compliant version of `strerror_r()`. Also, the only documented non-zero return values for `strerror_r()` I've found are `EINVAL` and `ERANGE` for an unrecognized error number and a too short buffer, respectively. However, in both cases the buffer should be filled with an error message that makes sense: an `Unknown error X` in the first case and a truncated message in the second. I think that ignoring the return value of `strerror_r()` is perfectly fine.

Reply:
So, if we ignore the return value of `strerror_r`, we don't need to distinguish between XSI version and GNU version. It simplifies the code to just three lines.